### DISCUSSION TITLE
CrazyRadio not able to connect to Scan with cfclient

### DISCUSSION THREAD

**Discussion by rowancllzz** [bitcraze/discussions#97] [Q&A] [answered]
Hi,

I recently just assembled my second ever CrazyFlie 2.0 with a multiranger deck on it. While after I finished assembling it and updating the firmware, it was able to scan and connect to my client when connecting it to my computer by means of a USB. After I turned it off and on again, I reattached the drone to my computer with a USB and it radio does not even scan this time.

I checked that the dongle is working fine. The first Crazyflie I built with a LPS deck is still able to scan and connect. However, with this drone, nothing happens. Currently, LEDs M2 and M3 are flashing while M1 blinks about 2 times a second (which seems like the power is on and everything is good).

I am not sure what to do. I was reading a lot of people with this problem will connect their Crazyflie to their client with a USB and configure it through settings. However, I do not get this option. 

Any help is appreciated.

**Comment by knmcguire** [comment]
Hi!

I need some more information about your problem first as there are some things that are unclear.

- Are you sure it is a Crazyflie 2.0 and not a Crazyflie 2.1?
- Which computer OS are you running the cfclient on?
- Which version of the cfclient are you running?
- It is a bit unclear from how you explain the situation. Are you:
  - or 1. Are you able to connect through it with USB and then try to use the Crazyradio PA to connect to it while connected through USB
  - or 2. Are you only trying to communicate through a USBcable?

**Reply by rowancllzz** [reply]
- Yes, it is a Crazyflie 2.0
- I am running the cfclient on Windows 10
- I am running on the cfclient GUI
- Number 1, I am trying to use the Crazyradio PA to connect it while connected through the USB. 

Please let me know if anything is still unclear. I am still relatively new with Crazyflie!

**Reply by knmcguire** [reply]
Thanks! Some more additional questions still I'm afraid, but hopefully we can figure out your issue!

- Did you update the firmware through the cfclient (2022.05) or did you flash the latest master of [Crazyflie-firmware](https://github.com/bitcraze/crazyflie-firmware)
- When you press 'scan', do you see the option `usb://0 `
- Is the other Crazyflie you mentioned also on when you try to attempt to scan?

**Reply by larasupernovae** [reply]
Hi! I seem to have the same problem only on Ubuntu 20.4, the usb port is working it is recognized. However when I try to scan and find the usb CrazyRadio it is not able to connect and cfclient is not working, this is the message I get in the shell:

INFO:cfclient.gui:Disabling STL printouts
INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']
INFO:cfclient.utils.input.inputreaders:Successfully initialized [linuxjsdev]
INFO:cfclient.utils.input.inputreaders:Could not initialize [pysdl2]: No SDL2 support on Linux
INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'
INFO:cfclient.utils.config:Dist config read from /home/larablackhole/crazyflie-clients-python/src/cfclient/configs/config.json
INFO:cfclient.utils.config:Config file read from [/root/.config/cfclient/config.json]
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file
INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in config file
INFO:cfclient.utils.zmq_param:ZMQ param disabled in config file
INFO:cfclient.gui:If startup fails because of "xcb", install dependency with `sudo apt install libxcb-xinerama0`.
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
INFO:cfclient.utils.input:Using device blacklist [(VirtualBox|VMware)]
INFO:cfclient.ui.tabs.QualisysTab:Switching Flight Mode to: FlightModeStates.DISCONNECTED
INFO:cflib.crtp.radiodriver:v0.53 dongle with serial N/A found
INFO:cflib.drivers.cfusb:Looking for devices....
WARNING:cflib.crtp.usbdriver:Exception while scanning for Crazyflie USB: 'NoneType' object has no attribute 'bcdDevice'
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.

Any help would be greatly appreciated.

**Reply by knmcguire** [reply]
@larasupernovae please start a new discussion for your situation. The problem here is with Windows and you have Ubuntu so it is probably a different issue. This makes it easier for us to help you and this thread does not get too complicated for the OP.

**Reply by larasupernovae** [reply]
@knmcguire No need for a new topic, it turned out that all of the crazyflies were damaged. Once I took a new one everything worked fine. Leaving this in case someone has a similar problem - IT WAS HARDWARE. Thank you!

**Comment by knmcguire** [comment]
@rowancllzz I'm just starting a new thread here :) but have you looked at the terminal's output from where the cfclient has started when you try to scan for your crazyflie? Could you copy paste that here?

**Reply by rowancllzz** [reply]
So it turns out the issue for me was the address. What I ended up doing was checking and rewriting the address to the Crazyflie via USB on the cfclient of my desktop computer. The cfclient there runs on the VM so was able to reconfigure it with a USB (for some reason you this doesn't work when the cfclient runs on Windows 10). Once I knew the address, I could type it in correctly and it would scan. Appreciate all your help!

**Reply by knmcguire** [reply]
Great! 

Could you post your eventual solution underneath here in 'suggest an answer' and then select 'mark as answer'? Then you can get kudos for being helpful and your solution will be placed on top right underneath your first question :)

**Reply by rowancllzz** [reply]
I'm sorry for not seeing this sooner. Will do it now!

**Comment by rowancllzz** [comment, marked as answer]
So it turns out the issue for me was the address. What I ended up doing was checking and rewriting the address to the Crazyflie via USB on the cfclient of my desktop computer. The cfclient there runs on the VM so was able to reconfigure it with a USB (for some reason you this doesn't work when the cfclient runs on Windows 10). Once I knew the address, I could type it in correctly and it would scan. Appreciate all your help!

**Reply by knmcguire** [reply]
Thanks! 

btw, I forgot to mention that for the usb cable on windows not detecting the crazyflie, you would need to configure it with Zadig as well, just like the crazyradio PA
